src/Toml/Type/Printer.hs

```diff
@@ -171,6 +175,19 @@ prettyKeyValue options i = mapOrdered (\kv -> [kvText kv]) options . HashMap.toL
     showText :: Show a => a -> Text
     showText = Text.pack . show
 
+    showTextUnicode :: Text -> Text
+    showTextUnicode text = Text.pack quotedText
+      where
+        quotedText = show finalText
+        finalText = foldl (\acc (ch, asciiCh) -> acc ++ getCh ch asciiCh) "" asciiArr 
+        xss = Text.unpack text
+        asciiArr = zip xss $ asciiStatus xss
+        getCh ch True  = [ch]
+        getCh ch False = printf "\\U%08x" ordChr :: String
+          where
+            ordChr = ord ch
+        asciiStatus = map isAscii
```

Comment:
Could you please add types to all functions and values inside the `where` block? Otherwise, it's a bit hard to understand the whole context and what is going on here.